nixos/lib/make-squashfs.nix

```diff
@@ -8,6 +8,9 @@
   comp ? "xz -Xdict-size 100%"
 }:
 
+let
+  compFlag = if comp == "none" then "-no-compression" else "-comp ${comp}";
```

Comment:
I think changing the type of `isoImage.squashfsCompression` to `nullOr str` as you described would be better since the string `none` isn't a valid `-comp` value.

Note that currently, the defaults and example for `isoImage.squashfsCompression` include _other parameters_ than `-comp`. For that reason, it might be even better to switch to a `listOf str` that are added as params, and rework the existing defaults to use that option, and make `squashfsCompression` a simple enum. In other words, you could use `squashfsCompression` to generate values for a new option like `squashfsExtraArgs`.

Reply:
Ah. At first I was going to protest since I'd still have to have a conditional for the flags, but I do see your point about composing the list of extra args as being more elegant/correct.

I guess my only concern is breaking other uses, but maybe that can be resolved.

Reply:
I went ahead and updated it to be `libs.types.nullOr lib.types.string` but I don't really want to do any larger changes right now.  I don't have the stomach for researching every enum value it can take, especially since it seems as you say it can take on arbitrary values depending on the compressor chosen.

Also, this change reminds fairly easy to grok, given that the implementation is in a separate nix file that takes its own parameters, and interpolates them into some bash script. I feel comfortable with this change and it seems like a net improvement. Someone could add more later if they wanted to further customize the squashfs builder.

Reply:
(Hope it's okay, I'm going to mark this as resolved, I did at least get rid of the stringy-"none" business. And I re-tested this change with `isoImage.compression = null` and testing it with my flake script that builds the ISO and tests it.)